www/front_src/src/App.tsx

```diff
@@ -64,154 +63,102 @@ const styles = createStyles({
-      });
-    }
-
-    // remove fullscreen parameters to keep normal routing
-    window.fullscreenSearch = null;
-    window.fullscreenHash = null;
+  const removeFullscreen = (): void => {
+    setIsFullscreenEnabled(false);
   };
 
-  // keep alive (redirect to login page if session is expired)
-  private keepAlive = (): void => {
-    this.keepAliveTimeout = setTimeout(() => {
-      axios('internal.php?object=centreon_keepalive&action=keepAlive')
-        .get()
-        .then(() => this.keepAlive())
-        .catch((error) => {
-          if (error.response && error.response.status === 401) {
-            // redirect to login page
-            window.location.href = 'index.php?disconnect=1';
-          } else {
-            // keepalive must be done cause it may failed due to temporary unavailability
-            this.keepAlive();
-          }
-        });
-    }, 15000);
+  const keepAlive = (): void => {
+    keepAliveRequest(keepAliveEndpoint).catch((error) => {
+      if (not(pathEq(['response', 'status'], 401, error))) {
+        return;
+      }
+
+      clearInterval(keepAliveIntervalRef.current as NodeJS.Timer);
+      window.location.href = './index.php?disconnect=1';
+    });
   };
 
-  public componentDidMount(): void {
-    // 2 - fetch external components (pages, hooks...)
-    this.props.fetchExternalComponents();
-
-    this.keepAlive();
-  }
-
-  public render(): ReactNode {
-    const min = this.getMinArgument();
-
-    const { classes } = this.props;
-
-    return (
-      <Suspense fallback={<PageLoader />}>
-        <ConnectedRouter history={history}>
-          <ThemeProvider>
-            <div className={classes.wrapper}>
-              {!min && <Nagigation />}
-              <div className={classes.content} id="content">
-                {!min && <Header />}
-                <div
-                  className={classes.fullScreenWrapper}
-                  id="fullscreen-wrapper"
-                >
-                  <Fullscreen
-                    enabled={this.state.isFullscreenEnabled}
-                    onChange={(isFullscreenEnabled): void => {
-                      this.setState({ isFullscreenEnabled });
-                    }}
-                    onClose={this.removeFullscreenParams}
-                  >
-                    <div className={classes.mainContent}>
-                      <MainRouter />
-                    </div>
-                  </Fullscreen>
-                </div>
-                {!min && <Footer />}
-              </div>
-              <Fab
-                className={classes.fullscreenButton}
-                color="default"
-                size="small"
-                onClick={this.goFull}
+  React.useEffect(() => {
+    getExternalComponents();
+    keepAlive();
+
+    keepAliveIntervalRef.current = setInterval(keepAlive, 15000);
+  }, []);
+
+  const min = hasMinArgument();
+
+  return (
+    <React.Suspense fallback={<PageLoader />}>
+      <ThemeProvider>
```

Comment:
Shouldn't we move the ThemeProvider to our Provider Component?

Reply:
ThemeProvider is already defined in the Provider component. I'll remove it